prerequisites/client._go

```diff
@@ -79,20 +79,45 @@ type service struct {
 }
 
 type CloseableTransport struct {
-	oauth2.Transport
+	http.RoundTripper
+}
+
+type idleConnectionsCloser interface {
+	CloseIdleConnections()
 }
 
 func (t *CloseableTransport) CloseIdleConnections() {
-	http.DefaultTransport.(*http.Transport).CloseIdleConnections()
+	base := t.RoundTripper
+	if o, ok := base.(*oauth2.Transport); ok {
+		// oath2.Transport doesn't implement what we need, but we know it stores
+		// the underlying RoundTripper in a field called Base and can access it
+		// here. It's likely an `http.Transport` and should implement the
+		// connection closer.
+		base = o.Base
+	}
+
+	if base == nil {
+		base = http.DefaultTransport
+	}
+
+	if closer, ok := base.(idleConnectionsCloser); ok {
+		closer.CloseIdleConnections()
+	} else {
+		panic("HTTP transport does not implement IdleConnectionsCloser")
+	}
 }
 
-func NewCloseableTransport(t *oauth2.Transport) *CloseableTransport {
-	return &CloseableTransport{
-		oauth2.Transport{
-			Base:   t.Base,
-			Source: t.Source,
-		},
+func NewCloseableTransport(t http.RoundTripper) *CloseableTransport {
+	// Attempt to fail fast if we get an invalid transport. Still have to check
+	// later as e.g. `oauth2.Transport.Base` could change at runtime.
+	if _, ok := t.(*oauth2.Transport); ok {
+		// Ok
+	} else if _, ok := t.(idleConnectionsCloser); ok {
+		// Ok
+	} else {
+		panic("HTTP transport does not implement IdleConnectionsCloser")
 	}
+	return &CloseableTransport{t}
```

Comment:
suggestion: I think it looks a little cleaner with a type switch, but completely up to you.
```suggestion
	switch t.(type) {
	case *oauth2.Transport, idleConnectionsCloser:
		// ok
	default:
		panic("HTTP transport does not implement IdleConnectionsCloser")
	}
	return &CloseableTransport{t}
```